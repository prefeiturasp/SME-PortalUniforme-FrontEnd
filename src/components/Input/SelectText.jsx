import PropTypes from "prop-types";
import React from "react";
import { InputErroMensagem } from "./InputErroMensagem";
import { HelpText } from "components/HelpText";
import { Dropdown } from "primereact/dropdown";
import "./style.scss";

export class SelectText extends React.Component {
  render() {
    const {
      className,
      disabled,
      esconderAsterisco,
      helpText,
      input,
      label,
      labelClassName,
      meta,
      options,
      name,
      placeholder,
      required
    } = this.props;
    return (
      <div className="input">
        {label && [
          required && !esconderAsterisco && (
            <span key={2} className="required-asterisk">
              *
            </span>
          ),
          <label
            key={1}
            htmlFor={name}
            className={`col-form-label ${labelClassName}`}
          >
            {label}
          </label>
        ]}
        <Dropdown
          {...input}
          options={options}
          autoWidth={false}
          className={`${className} ${meta.touched &&
            meta.error &&
            "invalid-field"}`}
          disabled={disabled}
          placeholder={placeholder}
          required={required}
        />
        <HelpText helpText={helpText} />
        <InputErroMensagem meta={meta} />
      </div>
    );
  }
}
SelectText.propTypes = {
  className: PropTypes.string,
  disabled: PropTypes.bool,
  esconderAsterisco: PropTypes.bool,
  helpText: PropTypes.string,
  input: PropTypes.object,
  label: PropTypes.string,
  labelClassName: PropTypes.string,
  meta: PropTypes.object,
  name: PropTypes.string,
  placeholder: PropTypes.string,
  required: PropTypes.bool,
  type: PropTypes.string,
  options: PropTypes.array
};

SelectText.defaultProps = {
  className: "",
  disabled: false,
  esconderAsterisco: false,
  helpText: "",
  input: {},
  label: "",
  labelClassName: "",
  meta: {},
  name: "",
  placeholder: "",
  required: false,
  type: "text",
  options: []
};
