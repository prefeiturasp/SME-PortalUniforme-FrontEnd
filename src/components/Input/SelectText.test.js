import { mount } from "enzyme";
import React from "react";
import { SelectText } from "./SelectText";

describe("test <SelectText>", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = mount(<SelectText />);
  });

  it("renders component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
