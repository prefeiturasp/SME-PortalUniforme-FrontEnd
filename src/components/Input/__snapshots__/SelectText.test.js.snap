// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <SelectText> renders component 1`] = `
<SelectText
  className=""
  disabled={false}
  esconderAsterisco={false}
  helpText=""
  input={Object {}}
  label=""
  labelClassName=""
  meta={Object {}}
  name=""
  options={Array []}
  placeholder=""
  required={false}
  type="text"
>
  <div
    className="input"
  >
    <Dropdown
      appendTo={null}
      ariaLabel={null}
      ariaLabelledBy={null}
      autoFocus={false}
      autoWidth={false}
      className=" undefined"
      dataKey={null}
      disabled={false}
      editable={false}
      filter={false}
      filterPlaceholder={null}
      id={null}
      inputId={null}
      itemTemplate={null}
      maxLength={null}
      onChange={null}
      onContextMenu={null}
      onMouseDown={null}
      optionLabel={null}
      options={Array []}
      panelClassName={null}
      panelStyle={null}
      placeholder=""
      required={false}
      scrollHeight="200px"
      showClear={false}
      style={null}
      tabIndex={null}
      tooltip={null}
      tooltipOptions={null}
      value={null}
    >
      <div
        className="p-dropdown p-component  undefined"
        id={null}
        onClick={[Function]}
        onContextMenu={null}
        onMouseDown={null}
        style={null}
      >
        <div
          className="p-hidden-accessible"
        >
          <input
            aria-label={null}
            aria-labelledby={null}
            disabled={false}
            id={null}
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            readOnly={true}
            role="listbox"
            tabIndex={null}
            type="text"
          />
        </div>
        <div
          className="p-hidden-accessible p-dropdown-hidden-select"
        >
          <select
            aria-hidden="true"
            required={false}
            tabIndex="-1"
          >
            <option
              value=""
            />
          </select>
        </div>
        <label
          className="p-dropdown-label p-inputtext p-dropdown-label-empty"
        >
          empty
        </label>
        <div
          className="p-dropdown-trigger"
        >
          <span
            className="p-dropdown-trigger-icon pi pi-chevron-down p-clickable"
          />
        </div>
        <DropdownPanel
          appendTo={null}
          filter={null}
          onClick={[Function]}
          panelClassName={null}
          panelStyle={null}
          scrollHeight="200px"
        >
          <div
            className="p-dropdown-panel p-hidden p-input-overlay"
            onClick={[Function]}
            style={null}
          >
            <div
              className="p-dropdown-items-wrapper"
              style={
                Object {
                  "maxHeight": "200px",
                }
              }
            >
              <ul
                className="p-dropdown-items p-dropdown-list p-component"
              />
            </div>
          </div>
        </DropdownPanel>
      </div>
    </Dropdown>
    <HelpText
      helpText=""
    >
      <div
        className="help-text"
      />
    </HelpText>
    <InputErroMensagem
      meta={Object {}}
    >
      <div
        className="error-or-warning-message"
      />
    </InputErroMensagem>
  </div>
</SelectText>
`;
